mdsuite/calculators/trajectory_calculator.py

```diff
@@ -226,6 +226,19 @@ def _handle_tau_values(self) -> np.array:
 
         return times
 
+    def _check_remainder(self):
+        """
+        Check that the remainder is compatible with the calculator.
+
+        It may come to pass that the remainder computed by the memory manager is not
+        divisible by your data range. In this case, it must be clipped such that it is.
+
+        Returns
+        -------
+        Updates the remainder attribute if required.
+        """
+        return self.remainder - (self.remainder % self.args.data_range)
+
```

Comment:
Yep, this needs to be cleaned basically. There was an option given to calculators to use a remainder and this is now gone. The checking here is now done within the ensemble generator which performs a check each ensemble that it can go through and if not the loop ends immediately.

Reply:
This stuff will be fully cleaned and re-built when we introduce ZnH5MD.